Added note regarding GHES support (#32)
actions/gh-actions-cache: README.md

This extension builds on top of [cache management](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#managing-caches) capabilities exposed by GitHub [APIs](https://docs.github.com/en/rest/actions/cache).

> **Note:** This extension currently supports github.com repositories only and will start supporting GHES (GitHub Enterprise Server) by the next ship version i.e. 3.7.

Reviewer: ```suggestion
**Note:** This extension currently supports github.com repositories only and will start supporting GHES (GitHub Enterprise Server) starting GHES 3.7.
```

Author: Removed this as we don't want to state the version as of now.

> **Note:** This extension currently supports github.com repositories only and will start supporting GHES (GitHub Enterprise Server) by the next ship version i.e. 3.7.

Author: Removed the sentence that states GHES and version and wrote `GitHub Enterprise Server is not supported currently.`

## Contributing

> If anything feels off, or if you feel that some functionality is missing, please check out the [contributing page](CONTRIBUTING.md). There you will find instructions for sharing your feedback, building the tool locally, and submitting pull requests to the project.

Reviewer 2: There is no change here right?

Author: No change. Could be a minor formatting by the editor.
